src/overlays/actors/ovl_En_Horse_Link_Child/z_en_horse_link_child.c

```diff
@@ -27,47 +31,558 @@ const ActorInit En_Horse_Link_Child_InitVars = {
+                             this->skin.skelAnime.animCurrentFrame,
+                             SkelAnime_GetFrameCount(&sAnimations[this->animationIdx]->genericHeader), 2, 0.0f);
+    }
+}
+
+void func_80A6A4DC(EnHorseLinkChild* this) {
+    this->action = 5;
+    this->animationIdx = Math_Rand_ZeroOne() > 0.5f ? 0 : 1;
+    DREG(53) = 0;
+    SkelAnime_ChangeAnim(&this->skin.skelAnime, sAnimations[this->animationIdx], func_80A695A4(this), 0.0f,
+                         SkelAnime_GetFrameCount(&sAnimations[this->animationIdx]->genericHeader), 2, 0.0f);
+}
+
+void func_80A6A5A4(EnHorseLinkChild* this, GlobalContext* globalCtx) {
+    s16 yawDiff;
+
+    if (DREG(53) != 0) {
+        DREG(53) = 0;
+        Audio_PlaySoundGeneral(NA_SE_EV_KID_HORSE_NEIGH, &this->actor.projectedPos, 4, &D_801333E0, &D_801333E0,
+                               &D_801333E8);
+        func_80A6A724(this);
+    } else {
+        this->actor.speedXZ = 0.0f;
+        yawDiff = func_8002DA78(&this->actor, &PLAYER->actor) - this->actor.posRot.rot.y;
+        // 0.7071 = cos(pi/4)
+        if ((Math_Coss(yawDiff) < 0.7071f) && (this->animationIdx == 2)) {
+            func_8006DD9C(&this->actor, &PLAYER->actor.posRot.pos, 300);
+        }
+
+        if (SkelAnime_FrameUpdateMatrix(&this->skin.skelAnime)) {
+            if (Math_Coss(yawDiff) < 0.0f) {
+                this->animationIdx = 2;
+                SkelAnime_ChangeAnim(
+                    &this->skin.skelAnime, sAnimations[this->animationIdx], D_80A6AF64[this->animationIdx], 0.0f,
+                    SkelAnime_GetFrameCount(&sAnimations[this->animationIdx]->genericHeader), 2, -5.0f);
+            } else {
+                func_80A6A4DC(this);
+            }
+        }
+    }
+}
+
+void func_80A6A724(EnHorseLinkChild* this) {
+    this->timer = 0;
+    this->action = 4;
+    this->animationIdx = 2;
+    this->unk_1E8 = false;
+    this->actor.speedXZ = 2.0f;
+    SkelAnime_ChangeAnim(&this->skin.skelAnime, sAnimations[this->animationIdx], func_80A695A4(this), 0.0f,
+                         SkelAnime_GetFrameCount(&sAnimations[this->animationIdx]->genericHeader), 2, -5.0f);
+}
+
+void func_80A6A7D0(EnHorseLinkChild* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    f32 dist;
+    s32 newAnimationIdx;
+
+    this->timer++;
+    if (this->timer > 300) {
+        this->unk_1E8 = true;
+    }
+
+    if ((this->animationIdx == 4) || (this->animationIdx == 3) || (this->animationIdx == 2)) {
+        if (!this->unk_1E8) {
+            func_8006DD9C(&this->actor, &player->actor.posRot.pos, 300);
+        } else {
+            func_8006DD9C(&this->actor, &this->actor.initPosRot.pos, 300);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse_Link_Child/func_80A695A4.s")
+    if (SkelAnime_FrameUpdateMatrix(&this->skin.skelAnime)) {
+        if (!this->unk_1E8) {
+            dist = func_8002DB8C(&this->actor, &PLAYER->actor);
```

Comment:
i assume PLAYER directly was necessary here

Reply:
It is. `PLAYER` is accessed twice and I had no luck with other combinations.